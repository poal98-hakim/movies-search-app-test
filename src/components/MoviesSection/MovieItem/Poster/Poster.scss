.poster{
    @media only screen and (max-width:750px) {
        height: 100%;
    }
}