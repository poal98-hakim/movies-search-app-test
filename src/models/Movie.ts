export interface IApiMovies{
    results: IMovie[];
    has_more: boolean;
    num_results: number;
}

export interface IMovie{
    display_title: string;
    mpaa_rating: string,
    critics_pick: number;
    byline: string,
    headline: string;
    summary_short: string;
    publication_date: string,
    opening_date: string;
    date_updated: string;
    link: ILink;
    multimedia: IMultimedia;
}

export interface ILink {
    type: string;
    url: string;
    suggested_link_text: string;
}

export interface IMultimedia {
    type: string;
    src: string;
    height: number;
    width: number;
}