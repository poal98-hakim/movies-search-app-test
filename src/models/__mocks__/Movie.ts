export const MOVIE1 = {
    display_title: "Everything Everywhere All at Once",
    mpaa_rating: "R",
    critics_pick: 1,
    byline: "A.O. Scott",
    headline: "‘Everything Everywhere All at Once’ Review: It’s Messy, and Glorious",
    summary_short: "Michelle Yeoh stars as a stressed-out laundromat owner dragged into cosmic battle and genre chaos.",
    publication_date: "2022-03-24",
    opening_date: "2022-03-25",
    date_updated: "2022-03-24 16:58:03",
    link: {
        type: "article",
        url: "https://www.nytimes.com/2022/03/24/movies/everything-everywhere-all-at-once-review.html",
        suggested_link_text: "Read the New York Times Review of Everything Everywhere All at Once"
    },
    multimedia: {
        type: "mediumThreeByTwo210",
        src: "https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg",
        height: 140,
        width: 210
    }
}
export const MOVIE2 = {
    display_title: "You Are Not My Mother",
    mpaa_rating: "R2",
    critics_pick: 1,
    byline: "Jeannette Catsoulis",
    headline: "‘You Are Not My Mother’ Review: Parental Misguidance",
    summary_short: "A lonely teenager is traumatized by her mother’s volatile behavior in this impressive horror debut.",
    publication_date: "2022-03-24",
    opening_date: "2022-03-25",
    date_updated: "2022-03-24 15:32:03",
    link: {
        type: "article",
        url: "https://www.nytimes.com/2022/03/24/movies/you-are-not-my-mother-review.html",
        suggested_link_text: "Read the New York Times Review of You Are Not My Mother"
    },
    multimedia: {
        type: "mediumThreeByTwo210",
        src: "https://static01.nyt.com/images/2022/03/25/arts/mother1/mother1-mediumThreeByTwo440.jpg",
        height: 140,
        width: 210
    }
}

export const API_MOVIES_RESULT = {
    results: [MOVIE1, MOVIE2],
    num_results: 2,
    hasMore: true
}