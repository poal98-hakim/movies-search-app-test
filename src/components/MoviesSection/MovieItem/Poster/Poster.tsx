import React from "react";
import { IMultimedia } from "../../../../models/Movie";
import "./Poster.scss";

interface IProps{
  multimedia: IMultimedia;
}

const Poster : React.VFC<IProps> = ({ multimedia }) => {
  const { src } = multimedia || {};
  return (
    <img className="poster full-width" src={src} alt="poster" />
  )
}

export default Poster;
