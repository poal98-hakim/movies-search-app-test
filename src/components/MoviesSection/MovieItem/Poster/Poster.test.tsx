import Poster from "./Poster";
import { render, screen } from '@testing-library/react';
import { API_MOVIES_RESULT } from "../../../../models/__mocks__/Movie"

const PROPS = {
  multimedia: API_MOVIES_RESULT.results[0].multimedia
}

describe('Poster', () => {
  let props = {...PROPS};

  beforeEach(() => { //before each test render the component and reset the value of props
    props = {...PROPS};
    render(<Poster {...props} />);
  });
  
  it('renders correctly', () => {
    const { baseElement } = render(<Poster {...props} />);
    expect(baseElement).toMatchSnapshot()
  })

  it('loads poster image', () => {
    expect(screen.getByAltText("poster")).toBeInTheDocument();
    expect(screen.getByAltText("poster")).toHaveAttribute("src", props.multimedia.src);
  })
  
})
