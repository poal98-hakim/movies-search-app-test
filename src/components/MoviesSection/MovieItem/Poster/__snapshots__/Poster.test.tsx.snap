// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster renders correctly 1`] = `
<body>
  <div>
    <img
      alt="poster"
      class="poster full-width"
      src="https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg"
    />
  </div>
  <div>
    <img
      alt="poster"
      class="poster full-width"
      src="https://static01.nyt.com/images/2022/03/24/arts/24everything-everywhere/merlin_204080367_730f3977-a2fa-425e-98db-01be8fecac21-mediumThreeByTwo440.jpg"
    />
  </div>
</body>
`;
